src/bsoncxx/document/value.cpp

```diff
@@ -53,6 +53,9 @@ value& value::operator=(const value& rhs) {
     return *this;
 }
 
+value::value(value&& rhs) noexcept: _data{std::move(rhs._data)}, _length{std::move(rhs._length)} {
```

Comment:
Another option would be to still revert that change, but try using clang's ASAN rather than GCCs? It might be interesting to see if you get better results compared to GCC for the stacks you don't understand yet?